src/task_5.py

```diff
@@ -0,0 +1,107 @@
+def checkio(f, g):
+    """Function validator
+
+    Two functions f and g are provided as inputs to checkio.
+    The first function f is the primary function and the second function g is
+    the backup. Use your coding skills to return a third function h which
+    returns the same output as f unless f raises an exception or returns None.
+    In this case h should return the same output as g. If both f and g raise
+    exceptions or return None, then h should return None.
+
+    As a second output, h should return a status string indicating whether
+    the function values are the same and if either function erred. A function
+    errs if it raises an exception or returns a null value (None).
+
+    The status string should be set to: "same" if f and g return the same
+    output and neither errs, "different" if f and g return different outputs
+    and neither errs, "f_error" if f errs but not g, "g_error" if g errs but
+    not f, or "both_error" if both err.
+
+    :param f: functions f (primary)
+    :param g: functions g (back up function)
+    :return: A function h which takes arbitrary inputs and returns a two-tuple
+    """
+    def h(*args, **kwargs):
```

Comment:
1. looks too complex but i don't see quick win for this for now
2. what if g returns None but doesn't raise exception?

Reply:
2. what if g returns None but doesn't raise exception?
if f_result is not None:
    if g_result is None:
                    return f_result, 'g_error'

if f_result == g_result and f_result is None:
            return None, 'both_error'